src/tools/chip-cert/CertUtils.cpp

```diff
@@ -1112,121 +1145,126 @@ CHIP_ERROR MakeCertTLV(CertType certType, const ToolChipDN * subjectDN, X509 * c
         writer.PutBytes(ContextTag(kTag_EllipticCurvePublicKey), subjectPubkey, chip::Crypto::CHIP_CRYPTO_PUBLIC_KEY_SIZE_BYTES));
 
     // extensions
-    ReturnErrorOnFailure(writer.StartContainer(ContextTag(kTag_Extensions), kTLVType_List, containerType2));
+    if (!useCompactIdentityFormat)
     {
-        if (isCA)
+        ReturnErrorOnFailure(writer.StartContainer(ContextTag(kTag_Extensions), kTLVType_List, containerType2));
         {
-            // basic constraints
-            if (certConfig.IsExtensionBasicPresent())
+            if (isCA)
             {
-                ReturnErrorOnFailure(writer.StartContainer(ContextTag(kTag_BasicConstraints), kTLVType_Structure, containerType3));
-                if (certConfig.IsExtensionBasicCAPresent())
-                {
-                    ReturnErrorOnFailure(writer.PutBoolean(ContextTag(kTag_BasicConstraints_IsCA),
-                                                           certConfig.IsExtensionBasicCACorrect() ? isCA : !isCA));
-                }
-                // TODO
-                if (pathLen != kPathLength_NotSpecified)
+                // basic constraints
+                if (certConfig.IsExtensionBasicPresent())
                 {
                     ReturnErrorOnFailure(
-                        writer.Put(ContextTag(kTag_BasicConstraints_PathLenConstraint), static_cast<uint8_t>(pathLen)));
+                        writer.StartContainer(ContextTag(kTag_BasicConstraints), kTLVType_Structure, containerType3));
+                    if (certConfig.IsExtensionBasicCAPresent())
+                    {
+                        ReturnErrorOnFailure(writer.PutBoolean(ContextTag(kTag_BasicConstraints_IsCA),
+                                                               certConfig.IsExtensionBasicCACorrect() ? isCA : !isCA));
+                    }
+                    // TODO
```

Comment:
What is the TODO here? Is there something obvious? If you can figure it out, would be great to either remove it, or document it. The original author is no longer working on Matter, so they are unlikely come back to this to make it clearer.

Reply:
No idea... will remove.